[APM][.NET] Add B3 documentation (#13727)
DataDog/documentation: content/en/tracing/setup_overview/setup/dotnet-core.md

: Enables improved resource names for web spans when set to `true`. Uses route template information where available, adds an additional span for ASP.NET Core integrations, and enables additional tags. Added in version 1.26.0. Enabled by default in 2.0.0<br>
**Default**: `true`

### Headers extraction and injection

> Datadog APM tracer supports [B3][9] and [W3C][10] headers extraction and injection for distributed tracing.

Reviewer: ```suggestion
The Datadog APM Tracer supports [B3][9] and [W3C][10] headers extraction and injection for distributed tracing.
```

> Distributed headers injection and extraction is controlled by configuring injection/extraction styles. Currently two styles are supported:

Reviewer 2: Datadog, B3, and W3C = 3?
```suggestion
Distributed headers injection and extraction is controlled by configuring injection/extraction styles. Currently three styles are supported:
```

Author: Wait 3? :mindblown: I'll learn how to count.
Fixed in 201b20b2e5bcf014ab433ee628af8fe84c7faf49 (i've actually set 4, cause I assume we should count B3 single as well

> Distributed headers injection and extraction is controlled by configuring injection/extraction styles. Currently four styles are supported:

Reviewer 2: Same as the other file :) 
```suggestion
Distributed headers injection and extraction is controlled by configuring injection/extraction styles. The .NET Tracer supports four styles:
```

> Distributed headers injection and extraction is controlled by configuring injection/extraction styles. The .NET Tracer supports four styles:

Reviewer: ```suggestion
You can configure injection and extraction styles for distributed headers. 

The .NET Tracer supports the following styles:
```

- Datadog: `Datadog`
- B3: `B3`
- W3C: `TraceParent`
- B3 Single Header: `B3SingleHeader` or `B3 single header`

> Injection and extraction styles can be configured using:

Reviewer: ```suggestion
You can use the following environment variables to configure injection and extraction styles:
```

> - Environment Variable: `DD_PROPAGATION_STYLE_INJECT=Datadog, B3, W3C`
> - Environment Variable: `DD_PROPAGATION_STYLE_EXTRACT=Datadog, B3, W3C`

Reviewer: ```suggestion
- `DD_PROPAGATION_STYLE_INJECT=Datadog, B3, W3C`
- `DD_PROPAGATION_STYLE_EXTRACT=Datadog, B3, W3C`
```

> The values of these environment variables are comma separated lists of header styles that are enabled for injection or extraction. By default only `Datadog` injection style is enabled.

Reviewer: ```suggestion
The environment variable values are comma-separated lists of header styles enabled for injection or extraction. By default, only the `Datadog` injection style is enabled.
```

> If multiple extraction styles are enabled extraction attempt is done on the order those styles are configured and first successful extracted value is used.

Reviewer: ```suggestion
If multiple extraction styles are enabled, the extraction attempt is completed in order of configured styles, and uses the first successful extracted value.
```
